src/libPMacc/PMaccConfig.cmake

```diff
@@ -76,37 +87,98 @@ foreach(PMACC_CUDA_ARCH_ELEM ${CUDA_ARCH})
     endif()
     unset(PMACC_IS_NUMBER)
 
-    # set flags to create device code for the given architecture
-    set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
-        "--generate-code arch=compute_${PMACC_CUDA_ARCH_ELEM},code=sm_${PMACC_CUDA_ARCH_ELEM} --generate-code arch=compute_${PMACC_CUDA_ARCH_ELEM},code=compute_${PMACC_CUDA_ARCH_ELEM}")
     if(${PMACC_CUDA_ARCH_ELEM} LESS 20)
         message(FATAL_ERROR "Unsupported CUDA architecture '${PMACC_CUDA_ARCH_ELEM}' specified. "
                             "Use '20' (for SM 2.0) or higher.")
     endif()
 endforeach()
 
-set(CUDA_FTZ "--ftz=false" CACHE STRING "Set flush to zero for GPU")
-
-set(CUDA_MATH --use_fast_math CACHE STRING "Enable fast-math" )
+option(CUDA_FAST_MATH "Enable fast-math" ON)
+option(CUDA_FTZ "Set flush to zero for CUDA" OFF)
 option(CUDA_SHOW_REGISTER "Show kernel registers and create PTX" OFF)
 option(CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps (folder: nvcc_tmp)" OFF)
-option(CUDA_SHOW_CODELINES "Show kernel lines in cuda-gdb and cuda-memcheck" OFF)
-
-if(CUDA_SHOW_CODELINES)
-    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" --source-in-ptx -Xcompiler -rdynamic -lineinfo)
-    set(CUDA_KEEP_FILES ON CACHE BOOL "activate keep files" FORCE)
-endif(CUDA_SHOW_CODELINES)
-
-set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} ${nvcc_flags} ${CUDA_MATH} ${CUDA_FTZ})
-if(CUDA_SHOW_REGISTER)
-    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" -Xptxas=-v)
-endif(CUDA_SHOW_REGISTER)
 
 if(CUDA_KEEP_FILES)
     file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/nvcc_tmp")
-    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" --keep --keep-dir "${PROJECT_BINARY_DIR}/nvcc_tmp")
 endif(CUDA_KEEP_FILES)
 
+if("${PMACC_CUDA_COMPILER}" STREQUAL "clang")
+    add_definitions(-DPMACC_CUDA_COMPILER_CLANG=1)
+    #set(LIBS ${LIBS} cudart_static)
+    set(CLANG_BUILD_FLAGS "-O3 -x cuda --cuda-path=${CUDA_TOOLKIT_ROOT_DIR}")
```

Comment:
Yes it is provided by [FindCUDA](https://cmake.org/cmake/help/v3.0/module/FindCUDA.html)